### PR TITLE
Extract drafts-related fields, methods, and properties from `Page` to `DraftStateMixin`

### PR DESCRIPTION
**Pull request by laymonage**
Needs #8531 to be merged.

_Please check the following:_

- [x] Do the tests still pass?[^1]
- [x] Does the code comply with the style guide? 
    - [x] Run `make lint` from the Wagtail root. 
- [x] For Python changes: Have you added tests to cover the new/fixed behaviour?

**Please describe additional details for testing this change**. 

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: wagtail/models/__init__.py [hunk 1]
```diff
@@ -386,7 +391,164 @@ class Meta:
...
+        """
+        Returns a new version of the object with field values updated to reflect changes
+        in the provided ``content`` (which usually comes from a previously-saved revision).
+
+        Certain field values are preserved in order to prevent errors if the returned
+        object is saved, such as ``id``. The following field values are also preserved,
+        as they are considered to be meaningful to the object as a whole, rather than
+        to a specific revision:
+
+        * ``latest_revision``
```
Comment (kaedroho) [inline wagtail/models/__init__.py:497]: Looks like this field is missing in the implementation

File: wagtail/models/__init__.py [hunk 2]
```diff
@@ -386,7 +391,164 @@ class Meta:
...
+        * ``live``
+        * ``has_unpublished_changes``
+        * ``first_published_at``
+
+        If ``TranslatableMixin`` is applied, the following field values are also preserved:
```
Comment (kaedroho) [inline wagtail/models/__init__.py:502]: I think we should delete this section from the docstring (this bit is implemented on ``Page``)

**Review by kaedroho — CHANGES_REQUESTED** [review]
Looks good to me, just some minor tweaks.

File: wagtail/models/__init__.py (hunk 2 above)
Comment (laymonage) [reply wagtail/models/__init__.py:502]: This `TranslatableMixin` bit is also implemented in `RevisionMixin`.. should we reword the docstring in here to state something like `in addition to the fields preserved in RevisionMixin, the following fields are also preserved`?

File: wagtail/models/__init__.py [hunk 3]
```diff
@@ -440,7 +440,7 @@ def _check_revision_mixin(cls):
             "DraftStateMixin requires RevisionMixin to be applied after DraftStateMixin.",
             hint="Add RevisionMixin to the model's base classes after DraftStateMixin.",
             obj=cls,
-            id="wagtail.EXXX",
+            id="wagtail.E004",
```
Comment (kaedroho) [inline wagtail/models/__init__.py:443]: ``wagtail`` => ``wagtailcore``

**Review by kaedroho — APPROVED** [review]

**Pull request merged**